documentation/Data Migration.md

```diff
@@ -0,0 +1,53 @@
+# Data Migration
+
+When upgrading to a new version of **Conforma** we may need to modify the existing database if the database schema has changed, or we've changed what data is stored in what fields.
+
+We need to account for both the following scenarios:
+
+1. Upgrading an existing server and migrate the data in place without taking a snapshot
+2. Loaded snapshots might require data migration before they can work correctly
+3. In development, we may need to update our dev data to work with code we're writing, even though the version number hasn't been increased yet.
+
+This process is handled by the "migrateData.ts" script, which runs automatically every time the server (re-)starts, or it can be run manually in the dev environment.
+
+## 1. Upgrading an existing server in place
+
+After [upgrading the server](Demo-Server-Guide.md), when the server restarts, the migration process is automatic. The script checks the version of the app saved in the database, compares it against the current (new) version number, and acts accordingly. If the saved version matches the current version, nothing happens. Each migration block has a version number check on it, and will run if the database version is less than this version number. So even when upgrading a server by several versions at once, the migration script will run in sequence for every previous version that has been skipped over.
+
+For live servers, data migration after an upgrade should be an invisible process.
+
+## 2. Migrating data with snapshots
+
```

Comment:
Maybe good to add here the existing limitation that `migration` doesn't run after importing the snapshot using the UI

Reply:
I've just simplified that whole paragraph in the docs to something much more generic -- there is a link to the issue if more detail is required. 1af63272

Though we should be fixing this soon, so will update docs properly then.